catroid/src/main/java/org/catrobat/catroid/content/bricks/SetInstrumentBrick.java

```diff
@@ -0,0 +1,220 @@
+		LEAD_8_BASS_AND_LEAD("Lead Bass and Lead", 88),
+		PAD_1_NEW_AGE("Pad New Age Fantasia", 89),
+		PAD_2_WARM("Pad Warm", 90),
+		PAD_3_POLYSYNTH("Pad Polysinth", 91),
+		PAD_4_CHOIR("Pad Choir Space Voice", 92),
+		PAD_5_BOWED("Pad Bowed Glass", 93),
+		PAD_6_METALLIC("Pad Metallic Pro", 94),
+		PAD_7_HALO("Pad Halo", 95),
+		PAD_8_SWEEP("Pad Sweep", 96),
+		FX_1_RAIN("FX Rain", 97),
+		FX_2_SOUNDTRACK("FX Soundtrack", 98),
+		FX_3_CRYSTAL("FX Crystal", 99),
+		FX_4_ATMOSPHERE("FX Atmosphere", 100),
+		FX_5_BRIGHTNESS("FX Brightness", 101),
+		FX_6_GOBLINS("FX Goblins", 102),
+		FX_7_ECHOES("FX 7 Echoes, Drops", 103),
+		FX_8_SCI_FI("FX Sci-fi", 104),
+		SITAR("Sitar", 105),
+		BANJO("Banjo", 106),
+		SHAMISEN("Shamisen", 107),
+		KOTO("Koto", 108),
+		KALIMBA("Kalimba", 109),
+		BAGPIPE("Bagpipe", 110),
+		SHANAI("Shanai", 112),
+		TINKLE_BELL("Tinkle bell", 113);
+
+		private String name;
+		private int value;
+		PickableMusicalInstrument(String name, int value) {
+			this.name = name;
+			this.value = value;
+		}
+
+		public static int getIndexByValue(int value) {
+			int index = 0;
+			for (PickableMusicalInstrument instrument : PickableMusicalInstrument.values()) {
+				if (instrument.getValue() == value) {
+					return index;
+				}
+				index++;
+			}
+			return -1;
+		}
+
+		@Override
+		public String getName() {
+			return name;
+		}
+
+		public int getValue() {
+			return value;
+		}
+
+		@Override
+		public void setName(String name) {
+			this.name = name;
+		}
+
+		public void setValue(int value) {
+			this.value = value;
+		}
+	}
+
+	public int instrumentSelection = PickableMusicalInstrument.values()[0].getValue();
+
+	@Override
+	public int getViewResource() {
+		return R.layout.brick_set_instrument;
+	}
+
+	protected int getSpinnerId() {
+		return R.id.set_instrument_spinner;
+	}
+
+	@Override
+	public View getView(Context context) {
+		super.getView(context);
+		List<Nameable> items = new ArrayList<>();
+
+		for (PickableMusicalInstrument instrument : PickableMusicalInstrument.values()) {
+			items.add(instrument);
+		}
+
+		BrickSpinner<PickableMusicalInstrument> spinner = new BrickSpinner<>(R.id.set_instrument_spinner, view, items);
+		spinner.setSelection(PickableMusicalInstrument.getIndexByValue(instrumentSelection));
+		spinner.setOnItemSelectedListener(new BrickSpinner.OnItemSelectedListener<PickableMusicalInstrument>() {
+
+			@Override
+			public void onStringOptionSelected(Integer spinnerId, String string) {
+			}
+
+			@Override
+			public void onItemSelected(Integer spinnerId, @Nullable PickableMusicalInstrument item) {
+				if (item != null) {
+					instrumentSelection = item.getValue();
+				}
+			}
+
+			@Override
+			public void onNewOptionSelected(Integer spinnerId) {
+			}
+		});
+		return view;
+	}
+
+	@Override
+	public void addActionToSequence(Sprite sprite, ScriptSequenceAction sequence) {
+		MusicalInstrument instrument =
+				MusicalInstrument.getInstrumentFromProgram(instrumentSelection - 1);
```

Comment:
since `MusicalInstrument.getInstrumentFromProgram()` is public static, please call it from the action.